DesignSafe-CI/portal
# Task/des 2676 add file folder button

## Overview: ##
Add Button is a drop-down with 5 clickable options: New Folder, New Project (not in this PR), File Upload, Folder Upload, and Bulk Data Transfer (link)

## PR Status: ##


## Related Jira tickets: ##

* [DES-2676](https://tacc-main.atlassian.net/browse/DES-2676)

## Summary of Changes: ##
- AddFIleFolder was created and used in DatafileBaseLayout
- AddFileFolder is a drop-down with 5 clickable options
- Three new modals were created: New Folder, Upload File, and Upload Folder

## Testing Steps: ##
1. New Folder
    - Modal pops up
    - If user clicks Create Folder with no name provided, an error message appears, 
    - File name can only contain alphanumeric characters, spaces, hyphens, underscores, periods, and parentheses.
    - As you move through folders, the path (in red) appears correctly
    - When a new folder is created, it appears in My Data
2. File Upload
    - Folder/Directory cannot be uploaded; only files in them
    - Reset and Start Upload are disabled if not files are selected
    - Multiple files can be selected at once
    - The value for staged files displays correctly
    - Reset button removes all staged files
    - When files are uploaded, they appear in My Data
3. Folder upload
    - Only one folder/directory can be selected, not individual files
    - Once a folder is selected, the user can remove individual files using the trash icon in the staged files list
    - The value for staged files displays correctly
    - Reset button removes all staged files
    - As you move through folders, the path (in red) appears correctly
    - When files are uploaded, they appear in My Data
4. Bulk Data Transfer
    - links to https://www.designsafe-ci.org/rw/user-guides/data-transfer-guide/

## UI Photos:
![Screen Shot 2024-03-20 at 1 19 36 PM](https://github.com/DesignSafe-CI/portal/assets/35277477/2588dfe8-5a2a-413d-ad4e-494a822318c6)
hotos:

## Notes: ##
